crates/primitives/oracle/src/lib.rs

```diff
@@ -0,0 +1,63 @@
+use anyhow::bail;
+use serde::{Deserialize, Serialize};
+
+mod pragma;
+
+use pragma::*;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(tag = "oracle_name", content = "config")]
+pub enum Oracle {
```

Comment:
Instead of using the Oracle enum for defining different oracle configurations, it might be more flexible and extensible to define a trait that represents the Oracle interface

Reply:
done